### PR TITLE
Sequence matcher

### PR DESCRIPTION
**Pull request by LassiInkinen**
Compute similarity ratio (converted to percentage) between every .sql exercise file.

### CODE REVIEW

File: src/sqlittaaja/printer.py [hunk 1]
```diff
@@ -1,18 +1,41 @@
...
-    print(f"┃ {{:^{max_name_len}}} ┃ {{}} ┃".format(name_col, score_col))
-    print(f"┡━{'━' * max_name_len}━╇━{'━' * len(score_col)}━┩")
-
-    # Each row.
-    for student_name, score in student_scores.items():
-        fmt = f"│ {{:<{max_name_len}}} │ {{:>{len(score_col)}}} │"
-        print(fmt.format(student_name, score))
-    print(f"└─{'─' * max_name_len}─┴─{'─' * len(score_col)}─┘")
+def print_scores(student_scores: dict[str, int], similarity_matrix: dict[tuple, str]):
+    # def print_scores(student_scores: dict[str, int]):
```
Comment (kuukkanen) [inline src/sqlittaaja/printer.py]: I guess this comment shouldn't be here?

File: src/sqlittaaja/printer.py [hunk 2]
```diff
@@ -16,3 +17,25 @@ def print_scores(student_scores: dict[str, int]):
         fmt = f"│ {{:<{max_name_len}}} │ {{:>{len(score_col)}}} │"
         print(fmt.format(student_name, score))
     print(f"└─{'─' * max_name_len}─┴─{'─' * len(score_col)}─┘")
+
+    """Prints similarity percentages."""
```
Comment (kuukkanen) [inline src/sqlittaaja/printer.py:21]: Use documentation comments only at the beginning of a function.

File: src/sqlittaaja/printer.py [hunk 3]
```diff
@@ -1,4 +1,5 @@
-def print_scores(student_scores: dict[str, int]):
+def print_scores(student_scores: dict[str, int], similarity_matrix: dict[tuple, str]):
+    # def print_scores(student_scores: dict[str, int]):
```
Comment (kuukkanen) [inline src/sqlittaaja/printer.py:2]: Remove this.

File: src/sqlittaaja/printer.py [hunk 4]
```diff
@@ -16,3 +17,25 @@ def print_scores(student_scores: dict[str, int]):
         fmt = f"│ {{:<{max_name_len}}} │ {{:>{len(score_col)}}} │"
         print(fmt.format(student_name, score))
     print(f"└─{'─' * max_name_len}─┴─{'─' * len(score_col)}─┘")
+
+    """Prints similarity percentages."""
+    if similarity_matrix:
```
Comment (kuukkanen) [inline src/sqlittaaja/printer.py:22]: Maybe integrate these into the pre-existing table printing?

File: src/sqlittaaja/diff_check.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    )
+    return similarity_ratio
+
+
+def compute_similarity(config: dict[str, Any]) -> dict[str, int]:
+    """Check students' exercises."""
+
+    similarity_matrix = {}
+
+    with ZipFile(config["exercise"]["path"], "r") as exercises:
```
Comment (kuukkanen) [inline src/sqlittaaja/diff_check.py:24]: We have already read and parsed the ZIP-file so should we maybe use the existing data here?

**Review by kuukkanen — CHANGES_REQUESTED** [review]

File: src/sqlittaaja/printer.py (hunk 4 above)
Comment (kuukkanen) [reply src/sqlittaaja/printer.py:22]: You could extract them into a more generic `print_table` function.

File: src/sqlittaaja/diff_check.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
+import difflib
+from typing import Any
+from zipfile import ZipFile
+from sqlittaaja.extractor import extract, student_info
+
+# Threshold percentage for warning >=
+threshold = 99
```
Comment (kuukkanen) [inline src/sqlittaaja/diff_check.py:7]: Maybe we should get this from the config and default to some value if not present?

**Review by kuukkanen — CHANGES_REQUESTED** [review]

File: src/sqlittaaja/diff_check.py (hunk 5 above)
Comment (LassiInkinen) [reply src/sqlittaaja/diff_check.py:24]: I had some original thought, that maybe the check could be run independently with some argument from cmd-line if needed, but propably not relevant atm.

**Pull request closed without merging**